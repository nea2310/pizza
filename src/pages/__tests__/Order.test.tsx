import { screen } from '@testing-library/react';
import Order from '../order/order';
import { renderWithProviders } from '../../shared/testUtils/testUtils';
import { initialStore } from '../../shared/testUtils/mockedStore';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';

// const mockedUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual<typeof import('react-router-dom')>('react-router-dom'),
  // useNavigate: () => mockedUseNavigate,
  // useLocation: jest.fn(() => {
  //   return { state: { total: 10 } };
  // }), - так не работает
  useLocation: () => {
    return { state: { total: 10 } };
  },
}));

describe('Order page rendering', () => {

  it('Renders Order page', async() => {
    const orderPage = renderWithProviders(<BrowserRouter><Order /></BrowserRouter>, {
      preloadedState: initialStore
    });

    expect(orderPage).toMatchSnapshot();

    const backToCart = screen.getByText(/Вернуться в корзину/);
    expect(backToCart).toBeInTheDocument();
    // fireEvent.click(backToCart);
    // expect(backToCart).toBeInTheDocument(); // не перешел в корзину
    const orderButton = screen.getByText(/Заказать/);
    expect(orderButton).toBeInTheDocument();
    const nameField = screen.getByLabelText(/Имя/);
    const phoneField = screen.getByLabelText(/Телефон/);
    const emailField = screen.getByLabelText(/Email/);
    expect(nameField).toBeInTheDocument();
    expect(phoneField).toBeInTheDocument();
    expect(emailField).toBeInTheDocument();
    userEvent.type(nameField, 'Тестовое Имя');
    userEvent.type(phoneField, '123456789');
    userEvent.type(emailField, 'gfhdsgfksfgfgftf@kjk.jhjh');
  });
});
