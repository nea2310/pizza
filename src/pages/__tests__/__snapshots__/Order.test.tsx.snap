// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order page rendering Renders Order page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="order"
      >
        <h1>
          Ваш заказ
        </h1>
        <hr />
        <form
          class="order__form"
        >
          <div>
            <label
              class="form-label"
              for="order-form-name"
            >
              Имя
            </label>
            <input
              class="form-control"
              id="order-form-name"
              type="text"
              value=""
            />
          </div>
          <div>
            <label
              class="form-label"
              for="order-form-phone"
            >
              Телефон
            </label>
            <input
              class="form-control"
              id="order-form-phone"
              type="text"
              value=""
            />
          </div>
          <div>
            <label
              class="form-label"
              for="order-form-email"
            >
              Email
            </label>
            <input
              class="form-control"
              id="order-form-email"
              type="text"
              value=""
            />
          </div>
        </form>
        <a
          class="order__button order__button_type_back"
          href="/cart"
        >
          Вернуться в корзину
        </a>
         
        <button
          class="order__button order__button_type_order"
          disabled=""
        >
          Заказать
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="order"
    >
      <h1>
        Ваш заказ
      </h1>
      <hr />
      <form
        class="order__form"
      >
        <div>
          <label
            class="form-label"
            for="order-form-name"
          >
            Имя
          </label>
          <input
            class="form-control"
            id="order-form-name"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            class="form-label"
            for="order-form-phone"
          >
            Телефон
          </label>
          <input
            class="form-control"
            id="order-form-phone"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            class="form-label"
            for="order-form-email"
          >
            Email
          </label>
          <input
            class="form-control"
            id="order-form-email"
            type="text"
            value=""
          />
        </div>
      </form>
      <a
        class="order__button order__button_type_back"
        href="/cart"
      >
        Вернуться в корзину
      </a>
       
      <button
        class="order__button order__button_type_order"
        disabled=""
      >
        Заказать
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
